src/Compilers/CSharp/Portable/Symbols/NullableAnnotationExtensions.cs

```diff
@@ -16,49 +19,75 @@ internal static class NullableAnnotationExtensions
+        /// they could be the same. These are the "invariant" merging rules.
         /// </summary>
-        public static NullableAnnotation EnsureCompatible(this NullableAnnotation a, NullableAnnotation b)
-        {
-            if (a.IsOblivious())
-                return b;
-            if (b.IsOblivious())
-                return a;
-            return (a < b) ? a : b;
-        }
+        public static NullableAnnotation EnsureCompatible(this NullableAnnotation a, NullableAnnotation b) =>
+            (a, b) switch
+            {
+                (NullableAnnotation.Oblivious, _) => b,
+                (_, NullableAnnotation.Oblivious) => a,
+                _ => a < b ? a : b,
+            };
 
         /// <summary>
         /// Merges nullability.
         /// </summary>
-        public static NullableAnnotation MergeNullableAnnotation(this NullableAnnotation a, NullableAnnotation b, VarianceKind variance)
-        {
-            return variance switch
+        public static NullableAnnotation MergeNullableAnnotation(this NullableAnnotation a, NullableAnnotation b, VarianceKind variance) =>
+            variance switch
             {
                 VarianceKind.In => a.Meet(b),
                 VarianceKind.Out => a.Join(b),
                 VarianceKind.None => a.EnsureCompatible(b),
                 _ => throw ExceptionUtilities.UnexpectedValue(variance)
             };
-        }
+
+        /// <summary>
+        /// The attribute (metadata) representation of <see cref="NullableAnnotation.NotAnnotated"/>.
+        /// </summary>
+        public const byte NotAnnotatedAttributeValue = 1;
+
+        /// <summary>
+        /// The attribute (metadata) representation of <see cref="NullableAnnotation.Annotated"/>.
+        /// </summary>
+        public const byte AnnotatedAttributeValue = 2;
+
+        /// <summary>
+        /// The attribute (metadata) representation of <see cref="NullableAnnotation.Oblivious"/>.
+        /// </summary>
+        public const byte ObliviousAttributeValue = 0;
+
+        /// <summary>
+        /// The mapping from NullableAnnotation to the attribute (metadata) representations.
+        /// </summary>
+        public static byte ToAttributeValue(this NullableAnnotation self) =>
+            self switch
+            {
+                NullableAnnotation.NotAnnotated => NotAnnotatedAttributeValue,
+                NullableAnnotation.Annotated => AnnotatedAttributeValue,
+                NullableAnnotation.Oblivious => ObliviousAttributeValue,
+                _ => throw ExceptionUtilities.UnexpectedValue(self)
+            };
+
+        ///// <summary>
+        ///// The mapping from the attribute (metadata) representations to NullableAnnotation.  This
+        ///// method is not actually used today, but is here to document the mapping.
+        ///// </summary>
+        //private static NullableAnnotation AttributeValueToNullableAnnotation(this byte value) =>
```

Comment:
>//private static NullableAnnotation AttributeValueToNullableAnnotation(this byte value) => [](start = 8, length = 90)

I think we have a policy to not check in commented out code, unless there is a bug opened to track that.

Reply:
I can add some random characters so it is no longer valid code, but that would just undermine the value of this comment.  This comment correctly describes the state of the compiler, so I prefer to keep it.

---
In reply to: [275055465](https://github.com/dotnet/roslyn/pull/34909#discussion_r275055465) [](ancestors = 275055465)

Reply:
> I can add some random characters so it is no longer valid code, but that would just undermine the value of this comment.  This comment correctly describes the state of the compiler, so I prefer to keep it.

I do not believe there is a value in keeping this code as a documentation. Comments for the new constants clearly describe the correspondence with values from ```NullableAnnotation``` enum. We can add similar notes to values in the ```NullableAnnotation``` enum as well. I think this will serve as much better than a comment in some random place that no one knows where to look for.

---
In reply to: [275056251](https://github.com/dotnet/roslyn/pull/34909#discussion_r275056251) [](ancestors = 275056251,275055465)